Bio/Seq.py

```diff
@@ -1280,6 +1281,54 @@ def ungap(self, gap=None):
             raise ValueError("Unexpected gap character, {0!r}".format(gap))
         return Seq(str(self).replace(gap, ""), alpha)
 
+    def join(self, other):
+        """Return a merge of the sequences in other, spaced by the sequence from self.
+
+        Accepts all Seq objects and Strings as objects to be concatenated with the spacer
+
+        >>> spacer = Seq('NNNNN')
+        >>> seqlist = [Seq("AAA"), Seq("TTT"), Seq("PPP")]
+        >>> concatenated = spacer.join(seqlist)
+        >>> concatenated
+        Seq('AAANNNNNTTTNNNNNPPP')
+
+        Throws error if other is not an iterable and if objects inside of the iterable
+        are not Seq or String objects
+        """
+        if not isinstance(other, collections.Iterable):  # doesn't detect single strings
+            raise ValueError("Input must be an iterable")
+        if isinstance(other, basestring):
```

Comment:
Should you also check for a single ``Seq`` object or ``MutableSeq`` object here? i.e.

```python
if isinstance(other, (basestring, Seq, MutableSeq)):
```

Reply:
This should already be accounted for in the `if not isinstance(other, collections.Iterable)`, as the test already has a line checking for a ValueError when a Seq object is an input on line 786 of test_Seq_objs.py: 
`self.assertRaises(ValueError, spacer.join, test_seq)`.
I can add another assert for MutableSeq.

Reply:
Ah - I was wrongly assuming the ``Seq`` etc objects would pass the Iterable check. Thanks.